plugins/modules/python_distribution.py

```diff
@@ -170,6 +191,15 @@ def main():
             else:
                 desired_attributes["remote"] = ""
 
+        if repository_name is not None:
+            if repository_name:
+                repository_ctx = PulpPythonRepositoryContext(
+                    module.pulp_ctx, entity={"name": repository_name}
+                )
+                desired_attributes["repository"] = repository_ctx.pulp_href
```

Comment:
Would you mind trying if this worked too?
```suggestion
                desired_attributes["repository"] = repository_ctx
```
I think glue should be able to resolve that.